pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,409 @@
+		nodeConfigCondition.Type = osev1.KubeletDegraded
+		nodeConfigCondition.Status = metav1.ConditionStatus(cond.Status)
+		nodeConfigCondition.Reason = "MachineConfigPoolRenderDegraded"
+		nodeConfigCondition.Message = "Machine Config Pool Render Degraded"
+	case mcfgv1.IsMachineConfigPoolConditionTrue(pool.Status.Conditions, mcfgv1.MachineConfigPoolNodeDegraded):
+		cond := mcfgv1.GetMachineConfigPoolCondition(pool.Status, mcfgv1.MachineConfigPoolNodeDegraded)
+		nodeConfigCondition.Type = osev1.KubeletDegraded
+		nodeConfigCondition.Status = metav1.ConditionStatus(cond.Status)
+		nodeConfigCondition.Reason = "MachineConfigPoolNodeDegraded"
+		nodeConfigCondition.Message = "Machine Config Pool Node Degraded"
+	case mcfgv1.IsMachineConfigPoolConditionTrue(pool.Status.Conditions, mcfgv1.MachineConfigPoolUpdated):
+		cond := mcfgv1.GetMachineConfigPoolCondition(pool.Status, mcfgv1.MachineConfigPoolUpdated)
+		nodeConfigCondition.Type = osev1.KubeletComplete
+		nodeConfigCondition.Status = metav1.ConditionStatus(cond.Status)
+		nodeConfigCondition.Reason = "MachineConfigPoolUpdated"
+		nodeConfigCondition.Message = "Machine Config Pool Updated"
+	case mcfgv1.IsMachineConfigPoolConditionTrue(pool.Status.Conditions, mcfgv1.MachineConfigPoolUpdating):
+		cond := mcfgv1.GetMachineConfigPoolCondition(pool.Status, mcfgv1.MachineConfigPoolUpdating)
+		nodeConfigCondition.Type = osev1.KubeletProgressing
+		nodeConfigCondition.Status = metav1.ConditionStatus(cond.Status)
+		nodeConfigCondition.Reason = "MachineConfigPoolUpdating"
+		nodeConfigCondition.Message = fmt.Sprintf("%d (ready %d) out of %d nodes are updating to latest configuration %s", pool.Status.UpdatedMachineCount, pool.Status.ReadyMachineCount, pool.Status.MachineCount, pool.Spec.Configuration.Name)
+	}
+	return ctrl.updateMCOStatus(nodeConfig, nodeConfigCondition)
+}
+
+// updateMCOStatus updates the MCO specific status condition present in the node config CR
+func (ctrl *Controller) updateMCOStatus(nodeConfig *osev1.Node, nodeConfigCondition metav1.Condition) error {
+	var (
+		index     int
+		isPresent bool
+	)
+	var kubeletConditionTypes = []string{
+		osev1.KubeletProgressing,
+		osev1.KubeletComplete,
+		osev1.KubeletDegraded,
+	}
+	for index = range nodeConfig.Status.WorkerLatencyProfileStatus.Conditions {
+		for _, conditionType := range kubeletConditionTypes {
+			if nodeConfig.Status.WorkerLatencyProfileStatus.Conditions[index].Type == conditionType {
+				isPresent = true
+				break
+			}
+		}
+		if isPresent {
+			break
+		}
+	}
+	if isPresent {
+		nodeConfig.Status.WorkerLatencyProfileStatus.Conditions[index] = *nodeConfigCondition.DeepCopy()
+	} else {
+		nodeConfig.Status.WorkerLatencyProfileStatus.Conditions = append(nodeConfig.Status.WorkerLatencyProfileStatus.Conditions, nodeConfigCondition)
+	}
+	_, err := ctrl.configClient.ConfigV1().Nodes().UpdateStatus(context.TODO(), nodeConfig, metav1.UpdateOptions{})
```

Comment:
Does anything else update the status? Could this be racy?

Reply:
This is the only place in the MCO where the NodeConfig status is getting updated